memory_reference/src/main.py

```diff
@@ -1,36 +1,34 @@
-from binding import *
+from binding import is_cuda_available
+from binding import is_hip_available
+from binding import get_available_device
+from binding import AlgoCPU
+import example_copy
 
 if __name__ == "__main__":
 
+    size = 10
+
     algos = [AlgoCPU()]
 
     if is_cuda_available() == True:
-        from algogpu import AlgoGPU
+        from algogpucuda import AlgoGPUCUDA
         import cupy_ref
-        algos.append(AlgoGPU())
+        algos.append(AlgoGPUCUDA())
 
+    if is_hip_available() == True:
+        from algogpuhip import AlgoGPUHIP
+        algos.append(AlgoGPUHIP())
+    
     print("Devices :")
     for device in get_available_device():
         print(device[0], end = ' ')
         if(device[1] == True) : print("Ready")
         else : print("Not Ready")
-    print()
-
-    for algo in algos:
-        
-        algo.whoami()
-        algo.initialize_array(10)
-
-        input = algo.get_input_memory()
-        output = algo.get_output_memory()
 
-        print(input)
-        print(type(input))
-
-        for i in range(10):
-            input[i] = 2.0
-
-        algo.compute(input, output)
+    print()
+    print("manual copy")
+    example_copy.example_manual_copy(algos, 10)
```

Comment:
I would prefer to move the functions `example_manual_copy` and `example_context_copy` to the `main.py`. The `main.py` is not really long and makes easier to read the core functionality.